src/main/java/com/zendesk/maxwell/RowMap.java

```diff
@@ -157,6 +157,32 @@ public String pkAsConcatString() {
 		return keys;
 	}
 
+	public String buildPartitionKey(List<String> partitionColumns, String partitionKeyFallback) {
+		String partitionKey="";
+		for (String pc : partitionColumns) {
+			Object pcValue = null;
+			if (data.containsKey(pc))
+				pcValue = data.get(pc);
+			if (pcValue != null)
+				partitionKey += pcValue.toString();
+		}
+		if (partitionKey.isEmpty())
+			return getPartitionKeyFallback(partitionKeyFallback);
+		return partitionKey;
+	}
+
+	private String getPartitionKeyFallback(String partitionKeyFallback) {
```

Comment:
@osheroff given this new config option should the pk fallback be configurable as well?